### PR TITLE
fix: Use importlib.resources to get in-process script

### PR DESCRIPTION
**Pull request by franzhaas**
uses importlib resources instead of __file__

### CODE REVIEW

**Review by frostming — APPROVED** [review]

File: src/pdm_packer/env.py [hunk 1]
```diff
@@ -14,7 +15,14 @@
...
     from pdm.models.environment import Environment as BaseEnvironment
 
-IN_PROCESS_SCRIPT = Path(__file__).with_name("_compile_source.py")
+
+def get_IN_PROCESS_SCRIPT():
+    try:
+        return (
+            importlib.resources.files("pdm_packer") / "_compile_source.py"
+        ).as_file()
+    except:
```
Comment (frostming) [inline src/pdm_packer/env.py:24]: Better to use a specific exception, or `sys.version_info >= (3, 9)`

Comment (franzhaas) [reply src/pdm_packer/env.py]: donne... did catch a mistake which made the code use the depricated path without me noticing...

File: src/pdm_packer/env.py [hunk 2]
```diff
@@ -14,7 +16,13 @@
 except ImportError:
     from pdm.models.environment import Environment as BaseEnvironment
 
-IN_PROCESS_SCRIPT = Path(__file__).with_name("_compile_source.py")
+
+def get_IN_PROCESS_SCRIPT():
```
Comment (frostming) [inline src/pdm_packer/env.py:20]:
```suggestion
def get_in_process_script():
```

File: src/pdm_packer/env.py [hunk 3]
```diff
@@ -33,8 +41,9 @@ def __exit__(self, *args: Any) -> None:
         self._dir.cleanup()
 
     def _compile_to_pyc(self, dest: Path) -> None:
-        args = [str(self.interpreter.path), str(IN_PROCESS_SCRIPT), str(dest)]
-        subprocess.check_output(args, stderr=subprocess.STDOUT)
+        with get_IN_PROCESS_SCRIPT() as scriptpath:
```
Comment (frostming) [inline src/pdm_packer/env.py:44]:
```suggestion
        with get_in_process_script() as scriptpath:
```

**Pull request merged**